Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -239,8 +238,8 @@ open class RouteController: NSObject {
     /// updateRouteLeg is used to notify nav-native of the developer changing the active route-leg.
     private func updateRouteLeg(to value: Int) {
         let legIndex = UInt32(value)
-        if navigator.changeRouteLeg(forRoute: 0, leg: legIndex), let timestamp = location?.timestamp {
-            updateIndexes(status: navigator.status(at: timestamp), progress: routeProgress)
+        if navigator.changeRouteLeg(forRoute: 0, leg: legIndex), let status = navigator.getStatus() {
```

Comment:
`Navigator.getStatus()` may be less safe than the preexisting call to `getStatusForMonotonicTimestampNanoseconds(_:)`, which we introduced in #2648 and relied on as a fix for #2583 and #2113. https://github.com/mapbox/mapbox-navigation-ios/issues/2846#issuecomment-824996423 proposed stashing away a copy of the status whenever we get it from NavigatorObserver callback; then we could use it directly and synchronously here.

Reply:
`Navigator.getStatus()` documentation states that it returns the last passed to NavigatorObserver status, so I don't think we should stash away a copy.